src/Extensions/Abstractions/NotNullWhenAttribute.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#if NETFRAMEWORK // This marker attribute only available starting from netstandard 2.1, so we need a replacement to build for full framework
+namespace System.Diagnostics.CodeAnalysis
+{
+	/// <summary>Specifies that when a method returns <see cref="ReturnValue"/>, the parameter will not be null even if the corresponding type allows it.</summary>
```

Comment:
Nitpick: `<summary>` tags should be on their own line as per .NET guidance

Reply:
I would really like to see this guidance :)
I will resolve it in the follow-up pr https://github.com/microsoft/Omex/pull/168

Reply:
The defaults and all examples from .NET use this format, and VS kind of forces you into it. I don't know if it's explicitly called out anywhere, but it always seems to be taken as a given.

Reply:
I understand, sorry for my rough expression. It's just I like to minimize space taken by comments as much as possible and this seemed as valid hack :)